Create a community-wide contributing guide in the repository (#26)
wiremock/community: contributing/README.md

# Contributing to WireMock

[![a](https://img.shields.io/badge/slack-Join%20us-brightgreen?style=flat&logo=slack)](http://slack.wiremock.org/)

WireMock exists and continues to thrive due to the efforts of over 150 contributors,
and we continue to welcome contributions to its evolution.
Regardless of your expertise and time you could dedicate,
there're opportunities to participate and help the project!

## Ways to contribute

You can help make WireMock a better tool in a number of ways:

- Help other WireMock users and sharing tips
- Write or improve documentation
- Raise an issue if you discover a bug
- Contribute bug fixes, new features or enhancements
- Spread the word about WireMock, share your stories and contributions

## Helping users

- Join the [community Slack channel](http://slack.wiremock.org/),
  especially the `#help` channel we use for random Q&A
- Subscribe to the [`#wiremock` tag on StackOverflow](https://stackoverflow.com/questions/tagged/wiremock).
  Many questions are raised there, and it is a good opportunity to get StackOverflow medals :smile:

> - If you have expertise with a particular WireMock component, e.g. _WireMock Java_ or _WireMock.NET_,

Reviewer: Let's leave .NET out of the picture until we've sorted out our relationship with that project.

Author: ```suggestion
- If you have expertise with a particular WireMock component, e.g. _WireMock Java_ or the _Python WireMock Client_,
```

Agreed

>   be sure maintainers will appreciate

Reviewer 2: Language is a little confusing - probably can do without `be sure`.

Author: Agreed!

```suggestion
   maintainers will appreciate
```

There are many repositories, and only a fraction of them, though the key ones,
are a part of the [WireMock organization](https://github.com/wiremock) on GitHub.
When contributing to WireMock, make sure to consult with the contributing guidelines of the repositories.

How to find a repository? There are a few options:

1. See the links below for key components
2. Consult with the [WireMock Ecosystem](https://github.com/wiremock/ecosystem) that provides links to the repositories
3. Ask in the `#help-contributing` channel

## Contributing to documentation

### WireMock Website

Most of WireMock's documentation is published on `wiremock.org`,
which is a static website built using Jekyll.
The website sources are located here: [wiremock/wiremock.org-sources](https://github.com/wiremock/wiremock.org-sources).
All the documentation is located under the `_docs` directory as Markdown files,
and it can be edited with all modern text editors and IDEs.

You can extend or modify documentation by submitting a pull request against the repository.
To do so, you can either fork and clone the repository and do edits in your branch,
or just use GitHub's web interface for small patches.

> ### Built-in docs

Reviewer: Can I suggest we make this section more specific about the API documentation i.e. where it is and how to update it?

And also drop the ref to Javadocs, since we don't really do them at the moment. We should consider whether they're all that useful for a library like this, and if so fix them up a bit before encouraging contributions.

Author: @tomakehurst For JavaDoc, we could agree to use https://javadoc.io/doc/com.github.tomakehurst/wiremock as the official location for now. Or I can quickly assemble publishing on https://wiremock.github.io/javadoc if we prefer to keep control in the community

Reviewer: If you feel strongly that we really need them, then let's keep them somewhere under wiremock.org

Author: Created https://github.com/wiremock/wiremock.org/issues/32 for the follow-up, will mention it from the doc

How to find a repository? There are a few options:

1. See the links below for key components
2. Consult with the [WireMock Ecosystem](https://github.com/wiremock/ecosystem) that provides links to the repositories
3. Ask in the `#help-contributing` channel

## Contributing to documentation

### WireMock Website

Most of WireMock's documentation is published on `wiremock.org`,
which is a static website built using Jekyll.
The website sources are located here: [wiremock/wiremock.org-sources](https://github.com/wiremock/wiremock.org-sources).
All the documentation is located under the `_docs` directory as Markdown files,
and it can be edited with all modern text editors and IDEs.

You can extend or modify documentation by submitting a pull request against the repository.
To do so, you can either fork and clone the repository and do edits in your branch,
or just use GitHub's web interface for small patches.

### Built-in docs

Some documentation is generated from code, for example Javadoc
This documentation can be (and should be) edited along with code so that it could be published later to the websites.

> - [Unofficial Javadoc site](https://javadoc.io/doc/com.github.tomakehurst/wiremock)

Author: ```suggestion
- [Unofficial Javadoc site](https://javadoc.io/doc/com.github.tomakehurst/wiremock)
  See [wiremock/wiremock.org #32](https://github.com/wiremock/wiremock.org/issues/32) for hosting an official Javadoc site
```
